src/strkey.rs

```diff
@@ -8,90 +10,41 @@ pub enum DecodeError {
 
 #[derive(Clone, Hash, PartialEq, Eq, Debug)]
 pub enum Strkey {
-    PublicKey(PublicKey),
+    PublicKeyEd25519([u8; 32]),
+    PrivateKeyEd25519([u8; 32]),
 }
 
 impl Strkey {
     pub fn to_string(&self) -> String {
         match self {
-            Self::PublicKey(x) => x.to_string(),
-        }
-    }
-
-    pub fn from_string(s: &str) -> Result<Self, DecodeError> {
-        let (ver, _) = decode(s)?;
-        match ver {
-            Version::PublicKeyEd25519 => Ok(Self::PublicKey(PublicKey::from_string(s)?)),
-        }
-    }
-}
-
-#[derive(Clone, Copy, Hash, PartialEq, Eq, Debug)]
-pub enum PublicKey {
-    Ed25519(PublicKeyEd25519),
-}
-
-impl PublicKey {
-    pub fn to_string(&self) -> String {
-        match self {
-            Self::Ed25519(x) => x.to_string(),
-        }
-    }
-
-    fn from_version_and_payload(ver: Version, payload: &[u8]) -> Result<Self, DecodeError> {
-        match ver {
-            Version::PublicKeyEd25519 => Ok(Self::Ed25519(
-                PublicKeyEd25519::from_version_and_payload(ver, payload)?,
-            )),
+            Self::PublicKeyEd25519(bs) => encode(Version::PublicKeyEd25519, bs),
+            Self::PrivateKeyEd25519(bs) => encode(Version::PrivateKeyEd25519, bs),
         }
     }
 
     pub fn from_string(s: &str) -> Result<Self, DecodeError> {
         let (ver, payload) = decode(s)?;
-        Self::from_version_and_payload(ver, &payload)
-    }
-}
-
-#[derive(Clone, Copy, Hash, PartialEq, Eq, Debug)]
-pub struct PublicKeyEd25519(pub [u8; 32]);
-
-impl PublicKeyEd25519 {
-    pub fn to_string(&self) -> String {
-        encode(Version::PublicKeyEd25519, &self.0)
-    }
-
-    fn from_version_and_payload(ver: Version, payload: &[u8]) -> Result<Self, DecodeError> {
+        let payload_slice = match <[u8; 32]>::try_from(payload) {
+            Ok(bs) => bs,
+            Err(_) => return Err(DecodeError::Invalid),
+        };
         match ver {
-            Version::PublicKeyEd25519 => match payload.try_into() {
-                Ok(ed25519) => Ok(Self(ed25519)),
-                Err(_) => Err(DecodeError::Invalid),
-            },
+            Version::PublicKeyEd25519 => Ok(Self::PublicKeyEd25519(payload_slice)),
+            Version::PrivateKeyEd25519 => Ok(Self::PrivateKeyEd25519(payload_slice)),
         }
     }
-
-    pub fn from_string(s: &str) -> Result<Self, DecodeError> {
-        let (ver, payload) = decode(s)?;
-        Self::from_version_and_payload(ver, &payload)
-    }
 }
 
-#[derive(Clone, Copy, Hash, PartialEq, Eq, Debug)]
+#[derive(Clone, Copy, Hash, PartialEq, Eq, Debug, TryFromPrimitive)]
 #[repr(u8)]
 enum Version {
     PublicKeyEd25519 = typ::PUBLIC_KEY | public_key_alg::ED25519,
-}
-
-impl Version {
-    fn try_from(b: u8) -> Result<Self, DecodeError> {
-        match b {
-            typ::PUBLIC_KEY | public_key_alg::ED25519 => Ok(Version::PublicKeyEd25519),
-            _ => Err(DecodeError::Invalid),
-        }
-    }
```

Comment:
I get the attraction with using the num_enum macro, but importing a couple thousands lines of complex macro code to replace 5 lines of a simple match statement seems overkill?

Reply:
ok. I begrudgingly agree. but this code hurts my eyes